WcaOnRails/app/views/registrations/register.html.erb

```diff
@@ -75,27 +77,28 @@
           <% end %>
         </div>
 
-        <%= simple_form_for @registration, url: competition_registrations_path(@competition.id), html: { class: 'form-horizontal are-you-sure no-submit-on-enter' }, wrapper: :horizontal_form,
+        <%= simple_form_for @registration, url: @registration.new_record? ? competition_registrations_path(@competition) : registration_path(@registration), html: { class: 'form-horizontal are-you-sure no-submit-on-enter' }, wrapper: :horizontal_form,
```

Comment:
Mhh, not sure. I haven't used SimpleForm much, but I have seen `simple_form_for [@competition, @registration]` in order to generate nested URLs.